lib/fast-path.js

```diff
@@ -316,6 +316,9 @@ FPp.needsParens = function(assumeExpressionContext) {
     case "ArrowFunctionExpression":
         return isBinary(parent);
 
+    case "ObjectExpression":
+        return parent.type === "ArrowFunctionExpression";
```

Comment:
Can you add `&& name === "body"` to the returned expression here? There's a small possibility that the object expression could be a default initializer expression, as in `(x, y={z:1}) => x + y.z`, and I don't think we'd want to force that `{z:1}` to be wrapped in parentheses.